test/functional/viml/errorlist_spec.lua

```diff
@@ -0,0 +1,49 @@
+local helpers = require('test.functional.helpers')
+
+local clear = helpers.clear
+local command = helpers.command
+local eq = helpers.eq
+local exc_exec = helpers.exc_exec
+local get_cur_win_var = helpers.curwinmeths.get_var
+
+describe('setqflist()', function()
+  local setqflist = helpers.funcs.setqflist
+
+  before_each(clear)
+
+  it('sets w:quickfix_title', function()
+    setqflist({''}, 'r', 'foo')
+    command('copen')
+    eq(get_cur_win_var('quickfix_title'), ':foo')
+  end)
+
+  it('expects a proper type for {title}', function()
+    command('copen')
+    setqflist({''}, 'r', '5')
+    eq(get_cur_win_var('quickfix_title'), ':5')
+    setqflist({''}, 'r', 6)
+    eq(get_cur_win_var('quickfix_title'), ':6')
+    local exc = exc_exec('call setqflist([""], "r", function("function"))')
+    eq(exc, 'Vim(call):E729: using Funcref as a String')
+    exc = exc_exec('call setqflist([""], "r", [])')
+    eq(exc, 'Vim(call):E730: using List as a String')
+    exc = exc_exec('call setqflist([""], "r", {})')
+    eq(exc, 'Vim(call):E731: using Dictionary as a String')
```

Comment:
AFAIR it is “expected, actual”, not “actual, expected”. This affects error message if test fails.